fsharp-backend/src/HttpMiddleware/HttpMiddlewareV1.fs

```diff
@@ -0,0 +1,134 @@
+/// Thin, bytes-friendly middleware
+///
+/// This Middleware corresponds with `HTTPBytes` handlers
+[<RequireQualifiedAccess>]
+module HttpMiddleware.HttpMiddlewareV1
+
+module Request =
+  open Prelude
+  open LibExecution.VendoredTablecloth
+
+  module RT = LibExecution.RuntimeTypes
+  module HttpQueryEncoding = BackendOnlyStdLib.HttpQueryEncoding
+
+  let private parseQueryString (query : string) : RT.Dval =
+    // Drop leading "?"
+    let query = if query.Length > 0 then String.dropLeft 1 query else query
+    HttpQueryEncoding.ofFormEncoding query
+
+  let private parseHeaders (headers : (string * string) list) =
+    headers
+    |> List.map (fun (k, v) -> (String.toLowercase k, RT.DStr v))
+    |> Map
+    |> RT.Dval.DObj
+
+  let private cookies (headers : HttpHeaders.T) : RT.Dval =
+    let parse (cookies : string) : RT.Dval =
+      let decode = System.Web.HttpUtility.UrlDecode
+      cookies
+      |> String.split ";"
+      |> List.map String.trim
+      |> List.map (fun s -> s.Split("=", 2) |> Array.toList)
+      |> List.map (fun cookie ->
+        match cookie with
+        | [] -> ("", RT.DNull) // skip empty rows
+        | [ _ ] -> ("", RT.DNull) // skip rows with only a key
+        | k :: v :: _ -> (decode k, RT.DStr(decode v)))
+      |> RT.Dval.obj
+
+    HttpHeaders.get "cookie" headers
+    |> Option.map parse
+    |> Option.defaultValue (RT.DObj Map.empty)
+
+  let private url (headers : List<string * string>) (uri : string) =
```

Comment:
Is this necessary in a minimal framework?

Reply:
Removed; FWIW [this](https://github.com/darklang/dark/blob/6ec452d4fc900198e761ca2fdee9404cff41ec9b/fsharp-backend/tests/httpbytestestfiles/url-custom-scheme.test) is one outcome of this logic being removed. Does that seem OK to you?

Reply:
I'm not sure what I'm looking at here, can you be more specific?

Reply:
In that test, the `X-forwarded-proto: ftp` is ignored.
I think that's pretty reasonable/expected, so probably not a useful question from me.